FindYourMovie/Controllers/MovieViewController.swift

```diff
@@ -0,0 +1,222 @@
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return movieSearch.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "MovieCell", for: indexPath) as! MovieCell
+        // Configure each cell
+        let movie = movieSearch[indexPath.row]
+        cell.name.text = movie.title
+        cell.releaseDate.text = movie.release_date
+        cell.overview.text = movie.overview
+        cell.poster.load(string: movie.poster_path)
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        let currentPosition = indexPath.row
+        let totalRows = movieSearch.count
+        // If the user has scrolled more than 90% of the list, then get the next available page, if available
+        if (self.currentPage < self.totalPages && Double(currentPosition)/Double(totalRows) >= 0.9) {
+            requestMovies(movieName: self.currentSearch, paging: true)
+        }
+    }
+
+    
+    /// Makes a call to the API requesting movies. It updates the internal values for total pages, current search, current page, and movies
+    ///
+    ///- Parameters:
+    ///     - movieName: name of the movie to search
+    ///     - paging:indicates if the request requires a page number. If true, the page number will be updated to the next available page.
+    /// if false, the request will be done to the first page
+    func requestMovies(movieName : String, paging: Bool = false)  {
+        
+        let movieNameformatted = movieName.replacingOccurrences(of: " ", with: "+")
+        UIApplication.shared.isNetworkActivityIndicatorVisible = true
+        
+        var pageNumber : Int
+        if paging {
+            pageNumber = self.currentPage + 1
+        } else {
+            pageNumber = 1
+        }
+        
+        MovieManager.shared.getMoviesNames(query: movieNameformatted, page: String(pageNumber))
+        { [weak self] results in
+            UIApplication.shared.isNetworkActivityIndicatorVisible = false
+            switch results {
+            case .success(let movieResult):
+                if pageNumber > 1 {
+                    self?.movieSearch.append(contentsOf: movieResult.0)
+                    self?.currentPage += 1
+                } else {
+                    self?.movieSearch = movieResult.0
+                    self?.totalPages = movieResult.1
+                    self?.currentSearch = movieNameformatted
+                    self?.currentPage = 1
+                    self?.updateLastSearchResults(movieName: movieNameformatted)
+                }
+                self?.tableView.reloadData()
+            case .failure(let error):
+                self?.showError(errorToDisplay : error)
```

Comment:
if on line ~93, we add
`guard let self = self else { return }` we can reference self throughout the selected with the `?`